runtime/artifact.go

```diff
@@ -49,17 +51,16 @@ func (context *TaskContext) UploadS3Artifact(artifact S3Artifact) error {
 		StorageType: "s3",
 	})
 	if err != nil {
-		return err
+		panic(errors.Wrap(err, "failed to Marshal json that should have worked"))
 	}
 
 	parsed, err := context.createArtifact(artifact.Name, req)
 	if err != nil {
 		return err
 	}
 	var resp queue.S3ArtifactResponse
-	err = json.Unmarshal(parsed, &resp)
-	if err != nil {
-		return err
+	if err = json.Unmarshal(parsed, &resp); err != nil {
+		panic(errors.Wrap(err, "failed to parse JSON that have been parsed before"))
```

Comment:
My "all input is evil" comment applies here too.

Reply:
`parsed` is cast from `json.RawMessage` in `context.createArtifact` so we know it is valid JSON.
hence, this is really just an assertion... it should **never** be possible for this to fail..